modules/engine/src/shader-inputs.ts

```diff
@@ -61,17 +61,20 @@ export class ShaderInputs<
    * @param modules
    */
   // @ts-expect-error Fix typings
-  constructor(modules: {[P in keyof ShaderPropsT]: ShaderModuleInputs<ShaderPropsT[P]>}) {
-    // TODO - get all dependencies from modules
-    const allModules = _resolveModules(Object.values(modules));
-    log.log(
-      1,
-      'Creating ShaderInputs with modules',
-      allModules.map(m => m.name)
-    )();
+  constructor(modules: {[P in keyof ShaderPropsT]?: ShaderModuleInputs<ShaderPropsT[P]>}) {
+    // Extract modules with dependencies and resolve
+    const allModules = {};
+    _getDependencyGraph({
```

Comment:
shadertools should export a function that just gets all module dependencies, we should not need to call such a. complicated internal function directly here. I was assuming _resolveModules did this, but apparently not.

Reply:
I used this because I wanted to avoid instantiating all the modules, I will add a helper function

Reply:
Instead, I've reverting to using `_resolveModules`, just with an added flag to disabled instantiation